### ISSUE TITLE
The "refwcs_NGC0253_DATACUBE_FINAL_*" files with wrong name.

### ISSUE THREAD

**Issue by cloud182** [emsellem/pymusepipe#11]
Hi @emsellem ,

While running finalise_reductin, I found another problem. When creating the datacube of the exposures and (I think) of the cubes, the pipeline expects a name of the tipe `refwcs_NGC0253_DATACUBE_FINAL_OBXXX` as you can see from the entry in the sof file:

`/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_OB001.fits OUTPUT_WCS`

However, the `refwcs_NGC0253_DATACUBE_FINAL_` are created with the old names:

```/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P01.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P02.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P03.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P04.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P05.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P06.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P07.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P08.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P09.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P10.fits
/data/beegfs/astro-storage/groups/schinnerer/PHANGS/MUSE/live/Data/PHANGS_EXT/NGC0253/Combined/Cubes/refwcs_NGC0253_DATACUBE_FINAL_P11.fits
[...]
```
I'll try to see if manually changing the name of the file can be a temporary fix.

Thank you in advance for your help!

E.

EDIT: Digging a little bit, in combine.py there are still a few recipes with "_P{:02d}" hardcoded in the file names.

**Comment by emsellem** [comment]
Nice catch! I'll see if I can fix this one.

**Comment by emsellem** [comment]
Actually, isn't it the sof which has the wrong entry? Shouldn't it be the P01, P02... and just that the OB0001 is wrong? Or?

**Comment by cloud182** [comment]
Hi Eric,

I'm not sure. I assumed the files were named wrong because they have the
same ID as the OB, and not as the pointings - i.e. they are P01, P02,
[...], P49, P998, P999 and not P01, [...], P99, P100.

But it could well be that they are correct and the sof is wrong. However, I
manually changed the names from PXX to OBXXX and everything seems to work.
The pipeline is now creating the exposure cubes, and the OB ones.

Cheers,
Enrico

Il mar 20 dic 2022, 06:59 emsellem ***@***.***> ha scritto:

**Comment by emsellem** [comment]
Ok. 
So I did change the P_{...} left-overs (odd as I thought I had fixed that).

But it won't change things because it should not be OB0001 etc but really the pointing numbers... So something is actually wrong in the way this is done. I believe the sof is fine but the naming is wrong since the refwcs should be corresponding to the Pointings... I believe that something is wrong. If you open one of the refwcs: does this use the pointing or the OBs? (I guess it will be the OBs as it will use all the expo in one OB).

This is the thing I have to do: make sense of the coherence as Pointings are only "part of the OBs" and there are more paintings that OBs. So the definition of the Pointings is the key here. And I am almost sure it's not implemented right yet.
